sw/device/silicon_creator/lib/drivers/spi_device.c

```diff
@@ -647,6 +647,9 @@ rom_error_t spi_device_cmd_get(spi_device_cmd_t *cmd) {
   reg = abs_mmio_read32(kBase + SPI_DEVICE_UPLOAD_STATUS2_REG_OFFSET);
   cmd->payload_byte_count =
       bitfield_field32_read(reg, SPI_DEVICE_UPLOAD_STATUS2_PAYLOAD_DEPTH_FIELD);
+  // Verify that the byte count in the PAYLOAD_DEPTH field does not exceed
+  // the actual hardware payload area buffer size.
+  HARDENED_CHECK_LE(cmd->payload_byte_count, kSpiDevicePayloadAreaNumBytes);
```

Comment:
One other thing to note that @nmosier pointed out is that on line 656 below that currently reads:
```c
for (size_t i = 0; i < cmd->payload_byte_count; i += sizeof(uint32_t)) {
```

if the `cmd->payload_byte_count` if not evenly divisible by 4 it could read up to 3 (potentially) garbage bytes into the buffer. Though I'm unfamiliar with the specifics of your implementation so this may be irrelevant. We just wanted to point this out in case.

Reply:
Thanks for checking @nmosier and @timothytrippel! This is the intended behavior. We prefer 32-bit reads for performance and since we already init the payload buffer area in spi_device_init() this out of range read does not cause any exceptions.

We *do* handle those bytes correctly in `bootstrap.c` by writing `0xff` instead of garbage.

Resolving, please re-open if you have any additional questions.